lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -223,3 +256,30 @@ export function updateDependency({
     return null;
   }
 }
+function overrideDepPosition(
+  overrideBlock: OverrideDependency,
+  parents: string[],
+  depName: string
+): {
+  depObjectReference: Record<string, string>;
+  overrideDepName: string;
+} {
+  // get override dep position when its nested in an object
+  const lastParent = parents[parents.length - 1];
+  let overrideDep: OverrideDependency = overrideBlock;
+  for (const parent of parents) {
+    if (overrideDep) {
+      overrideDep = overrideDep[parent]! as Record<string, RecursiveOverride>;
+    }
+  }
+  const overrideDepName = depName === lastParent ? '.' : depName;
+  const depObjectReference = overrideDep as Record<string, string>;
+  return { depObjectReference, overrideDepName };
+}
+
+function isOverrideObject(upgrade: Upgrade<NpmManagerData>): boolean {
+  return (
+    !is.undefined(upgrade.managerData?.parents) &&
```

Comment:
```suggestion
    is.array(upgrade.managerData?.parents, is.nonEmptyStringAndNotWhitespace) &&
```

- https://github.com/sindresorhus/is#arrayvalue-assertion
- https://github.com/sindresorhus/is#nonemptystringvalue